core/src/main/java/com/datastax/oss/driver/internal/core/metrics/DropwizardMetricsFactory.java

```diff
@@ -60,19 +61,19 @@ public DropwizardMetricsFactory(InternalDriverContext context) {
       LOG.debug("[{}] All metrics are disabled, Session.getMetrics will be empty", logPrefix);
       this.registry = null;
       this.sessionUpdater = NoopSessionMetricUpdater.INSTANCE;
-      this.metrics = Optional.empty();
+      this.metrics = null;
     } else {
       this.registry = new MetricRegistry();
       DropwizardSessionMetricUpdater dropwizardSessionUpdater =
           new DropwizardSessionMetricUpdater(enabledSessionMetrics, registry, context);
       this.sessionUpdater = dropwizardSessionUpdater;
-      this.metrics = Optional.of(new DefaultMetrics(registry, dropwizardSessionUpdater));
+      this.metrics = new DefaultMetrics(registry, dropwizardSessionUpdater);
     }
   }
 
   @Override
   public Optional<Metrics> getMetrics() {
-    return metrics;
+    return Optional.ofNullable(metrics);
```

Comment:
Why not, I don't care either way since this method should not be called very often.
But for my understanding: any rationale for wrapping at the last minute rather of storing the optional as a field?

Reply:
I found a few other occurrences of `Optional` as instance fields. I will fix them in a separate commit, but if you prefer to keep things the way they are, we can just drop that commit later.

Reply:
I think if the field is final, the enclosing type is not meant to be serializable, and it is on the hot path, then we should store the optional as a field. Otherwise, I'm fine recreating it each time.
(so this particular example falls into the "rewrap on every call" category)